crates/goose-server/src/auth.rs

```diff
@@ -0,0 +1,22 @@
+use axum::{
+    extract::{Request, State},
+    http::StatusCode,
+    middleware::Next,
+    response::Response,
+};
+
+pub async fn layer_fn(
```

Comment:
is this the normal way to name these? check_token would be easier to follow

Reply:
Nope it's just a bad name